config/v1/types_feature.go

```diff
@@ -40,6 +40,9 @@ var (
 
 	// IPv6DualStackNoUpgrade enables dual-stack. Turning this feature set on IS NOT SUPPORTED, CANNOT BE UNDONE, and PREVENTS UPGRADES.
 	IPv6DualStackNoUpgrade FeatureSet = "IPv6DualStackNoUpgrade"
+
+	// CSIMigrationNoUpgrade enables CSIMigration support. Turning this feature set on IS NOT SUPPORTED, CANNOT BE UNDONE, and PREVENTS UPGRADES.
+	CSIMigrationNoUpgrade FeatureSet = "CSIMigrationNoUpgrade"
```

Comment:
I'd prefer to see some enhancement with a bigger picture how the flags are going to be used.

In Kubernetes, CSI migration is indeed controller by setting some alpha feature gates, but they must be set in a certain order - control plane first (mainly kube-controller-manager and scheduler) and only after all their replicas are restarted and know about the new flag, nodes must be drained, have the feature flipped and uncordoned.
The other way around when disabling the feature - nodes first, once all of them are done (how do you detect that?), you can flip the control plane.

Does setting FeatureGate ensures this behavior? Can it ensure it? If now, then FeatureGate is probably not the right vehicle for CSI migration and something else must be used. In other words, please write an enhancement that would cover CSI migration enablement / disablement end-to-end and then we can file PRs for new APIs. In addition, it should work together with disabling in-tree cloud providers - as longs as CSI migration is disabled in kube-controller-manager, in-tree volume plugin + corresponding cloud provider must be enabled & working.